x/nft/keeper/nft.go

```diff
@@ -0,0 +1,192 @@
+	iterator := nftStore.Iterator(nil, nil)
+	defer iterator.Close()
+	for ; iterator.Valid(); iterator.Next() {
+		var nft nft.NFT
+		k.cdc.MustUnmarshal(iterator.Value(), &nft)
+		nfts = append(nfts, nft)
+	}
+	return nfts
+}
+
+// GetOwner returns the owner information of the specified nft
+func (k Keeper) GetOwner(ctx sdk.Context, classID string, nftID string) sdk.AccAddress {
+	store := ctx.KVStore(k.storeKey)
+	bz := store.Get(ownerStoreKey(classID, nftID))
+	return sdk.AccAddress(bz)
+}
+
+// GetBalance returns the specified account, the number of all nfts under the specified classID
+func (k Keeper) GetBalance(ctx sdk.Context, classID string, owner sdk.AccAddress) uint64 {
+	nfts := k.GetNFTsOfClassByOwner(ctx, classID, owner)
+	return uint64(len(nfts))
+}
+
+// GetTotalSupply returns the number of all nfts under the specified classID
+func (k Keeper) GetTotalSupply(ctx sdk.Context, classID string) uint64 {
+	store := ctx.KVStore(k.storeKey)
+	bz := store.Get(classTotalSupply(classID))
+	return sdk.BigEndianToUint64(bz)
+}
+
+// HasNFT determines whether the specified classID and nftID exist
+func (k Keeper) HasNFT(ctx sdk.Context, classID, id string) bool {
+	store := k.getNFTStore(ctx, classID)
+	return store.Has([]byte(id))
+}
+
+func (k Keeper) setNFT(ctx sdk.Context, token nft.NFT) {
+	nftStore := k.getNFTStore(ctx, token.ClassId)
+	bz := k.cdc.MustMarshal(&token)
+	nftStore.Set([]byte(token.Id), bz)
+}
+
+func (k Keeper) setOwner(ctx sdk.Context, classID, nftID string, owner sdk.AccAddress) {
+	store := ctx.KVStore(k.storeKey)
+	store.Set(ownerStoreKey(classID, nftID), owner.Bytes())
+
+	ownerStore := k.getClassStoreByOwner(ctx, owner, classID)
+	ownerStore.Set([]byte(nftID), Placeholder)
+}
+
+func (k Keeper) deleteOwner(ctx sdk.Context, classID, nftID string, owner sdk.AccAddress) {
+	store := ctx.KVStore(k.storeKey)
+	store.Delete(ownerStoreKey(classID, nftID))
+
+	ownerStore := k.getClassStoreByOwner(ctx, owner, classID)
+	ownerStore.Delete([]byte(nftID))
+}
+
+func (k Keeper) getNFTStore(ctx sdk.Context, classID string) prefix.Store {
+	store := ctx.KVStore(k.storeKey)
+	return prefix.NewStore(store, nftStoreKey(classID))
+}
+
+func (k Keeper) getClassStoreByOwner(ctx sdk.Context, owner sdk.AccAddress, classID string) prefix.Store {
+	store := ctx.KVStore(k.storeKey)
+	key := nftOfClassByOwnerStoreKey(owner, classID)
+	return prefix.NewStore(store, key)
+}
+
+func (k Keeper) incrTotalSupply(ctx sdk.Context, classID string) {
+	store := ctx.KVStore(k.storeKey)
+	supplyKey := classTotalSupply(classID)
+	bz := store.Get(supplyKey)
+	supply := sdk.BigEndianToUint64(bz) + 1
+	store.Set(supplyKey, sdk.Uint64ToBigEndian(supply))
+}
+
+func (k Keeper) decrTotalSupply(ctx sdk.Context, classID string) {
+	store := ctx.KVStore(k.storeKey)
+	supplyKey := classTotalSupply(classID)
+	bz := store.Get(supplyKey)
+	supply := sdk.BigEndianToUint64(bz) - 1
+	store.Set(supplyKey, sdk.Uint64ToBigEndian(supply))
+}
```

Comment:
maybe we can have a one more private function: `updateTotalSupply(ctx sdk.Context, classID string, num int64)` and use it in both functions above?